### PR TITLE
V7 Maintenance

### PR DESCRIPTION
**Pull request by squidadm**
- Branch 7.0.0
- Source Format Enforcement

### CODE REVIEW

**Comment by yadij** [comment]
OK to test

**Review by rousskov — COMMENTED** [review]
Thank you for updating these files.

File: tools/squidclient/squidclient.cc [hunk 1]
```diff
@@ -441,7 +441,7 @@ main(int argc, char *argv[])
         if (const auto at = strrchr(url, '@')) {
             if (!OriginAuthorization.user) {
                 std::cerr << "ERROR: Embedding a password in a cache manager command requires " <<
-                    "providing a username with -U: mgr:" << t << std::endl;
+                          "providing a username with -U: mgr:" << t << std::endl;
                 exit(EXIT_FAILURE);
```
[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: Thats the nice part. **none** of this was manual.

File: configure.ac [hunk 2]
```diff
@@ -5,7 +5,7 @@
 ## Please see the COPYING and CONTRIBUTORS files for details.
 ##
 
-AC_INIT([Squid Web Proxy],[6.0.0-VCS],[http://bugs.squid-cache.org/],[squid])
+AC_INIT([Squid Web Proxy],[7.0.0-VCS],[http://bugs.squid-cache.org/],[squid])
```
[the remark this answers is not in the document]
Comment (yadij) [reply configure.ac:8]: The manual changes to add release-7.* file are in [PR #1264 ](https://github.com/squid-cache/squid/pull/1264).
The script doing the "Branch N.0.0" commit is updated by https://github.com/squid-cache/ci/pull/9.

**Review by yadij — APPROVED** [review]

File: tools/squidclient/squidclient.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: The essence of my suggestion applies to mixing two unrelated automated changes[^1] just like it applies to mixing manual and automated changes, but, fortunately, we do not need to agree on the application of that principle in this PR context.

[^1]:  especially when one of those changes is fairly important

File: configure.ac (hunk 2 above)
[the remark this answers is not in the document]
Comment (rousskov) [reply configure.ac]: > The manual changes to add release-7.* file are in [PR #1264 ](https://github.com/squid-cache/squid/pull/1264).

AFAICT, version-neutral #1264 does not (and should not!) add a `release-7.*` (or similar) files. *This* PR should do that.

If the current (pretty much irrelevant here) scripts are not yet capable of creating that file, this "V7" PR should add it "manually". It is the effect of this PR that should be driving our decisions here, not the current state of some scripts. We should strive for an "atomic" (i.e. single commit) switch to a new future version in the master branch, and this PR ought to create such a switch.

File: tools/squidclient/squidclient.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: As I said, this was fully automated. What you are asking for is a major change to the maintenance scripts and to how the branch maintainers do their work. We are nowhere near ready for that yet. Meanwhile lack of these changes is now holding up daily snapshots.

[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: > What you are asking for is a major change to the maintenance scripts and to how the branch maintainers do their work.

This change request does not ask for any changes to the maintenance scripts or to how the branch maintainers do their work. This change requests suggests splitting one PR mixing two very different changes in two PRs.

File: configure.ac (hunk 2 above)
[the remark this answers is not in the document]
Comment (yadij) [reply configure.ac]: That has been fixed. PR #1264 has been amended.

File: tools/squidclient/squidclient.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: As I have said repeatedly this "one PR" is fully automated.

To be clear:
 * If we simply closed this PR right now. A new identical one would be opened in 5-6 hrs.

[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: > As I have said repeatedly this "one PR" is fully automated.

I am reviewing the results of that alleged automation. They should be improved, one way or the other. IMO, we cannot address or even justify the problems in this PR by simply saying that these problems were created by some scripts!

> If we simply closed this PR right now. A new identical one would be opened in 5-6 hrs.

I do not see the relevance of that assertion to this change request. This change request is not asking to close this PR. Obviously, humans can stop automation from generating new defective PRs or work around known automation limitations by doing something manually. To address this specific change request, we can, for example, isolate the automated `tools/squidclient/squidclient.cc` formatting changes into its own PR. If you want me to do that, just ask.

File: configure.ac (hunk 2 above)
[the remark this answers is not in the document]
Comment (rousskov) [reply configure.ac]: > Change request: This change is missing `doc/release-notes/release-7.sgml` (or similar) addition

> Alex: Version-neutral PR 1264 does not (and should not!) add a `release-7.*` (or similar) files. This PR should do that.

> Amos: That has been fixed. PR 1264 has been amended.

Amending PR 1264 does the _opposite_ of what is being requested here.

File: tools/squidclient/squidclient.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: So be it. Squid-7 branching and v6.0.1 are rolled back out of existence.

[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: FWIW, I do not understand why this PR got closed.

[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: github automatically closes PRs for removed code branches.

[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: FWIW, I do not understand why this PR branch was removed (triggering PR closure).

[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: "[Branch 7.0.0](https://github.com/squid-cache/squid/pull/1268/commits/6e305f7a0346a783229cfeffff3f01ffb06623bd)" was rolled back.

[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: FWIW, I do not understand why those changes were rolled back.

[the remark this answers is not in the document]
Comment (yadij) [reply tools/squidclient/squidclient.cc:445]: > This change requests suggests splitting one PR mixing two very different changes into two PRs.

[the remark this answers is not in the document]
Comment (rousskov) [reply tools/squidclient/squidclient.cc:445]: While one of those "two PRs" could have been this PR (i.e. there was no need or even a suggestion to close it, quite the [opposite](https://github.com/squid-cache/squid/pull/1268#discussion_r1101869462)), it is, of course, _your_ call how to address this change request. I just wanted to understand why this PR got closed (especially after we have invested so much time into it), and now I do. Thank you. Looking forward to those two followup PRs.

**Pull request closed without merging**